lib/internal/util/inspect.js

```diff
@@ -495,18 +495,23 @@ function formatValue(ctx, value, recurseTimes, typedArray) {
     return ctx.stylize('null', 'null');
   }
 
+  // Memorize the context for custom inspection on proxies.
+  const context = value;
+  // Always check for proxies to prevent side effects and to prevent triggering
+  // any proxy handlers.
+  const proxy = getProxyDetails(value);
```

Comment:
I did (see https://github.com/nodejs/node/pull/26241#issuecomment-466047172).

The default case is not going to slow down at all as this function first checks if the object is indeed a proxy or not. That check does not cause any performance penalty.

I did not benchmark objects that contain proxies and by how much they might slow down or not. I can still do that but I do not see it as a big issue as it's not the typical case.

The performance of inspect immensely improved since the original implementation and also proxy inspection itself got significantly faster (if I remember correct by 400%).